error[E0308]: mismatched types
  --> $DIR/bad-item-type.rs:13:11
   |
13 |     yield val; //~ ERROR mismatched types
   |           ^^^
   |           |
   |           expected enum `std::option::Option`, found integer
   |           help: try using a variant of the expected enum: `Some(val)`
   |
   = note: expected enum `std::option::Option<_>`
              found type `{integer}`

error[E0698]: type inside generator must be known in this context
 --> $DIR/bad-item-type.rs:7:9
  |
7 |     let val = Some(42);
  |         ^^^ cannot infer type for type `{integer}`
  |
note: the type is part of the generator because of this `yield`
 --> $DIR/bad-item-type.rs:5:1
  |
5 | #[stream(item = Option<i32>)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:12:9
   |
12 |     let val = val.unwrap();
   |         ^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:5:1
   |
5  | #[stream(item = Option<i32>)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:13:11
   |
13 |     yield val; //~ ERROR mismatched types
   |           ^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:5:1
   |
5  | #[stream(item = Option<i32>)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> $DIR/bad-item-type.rs:32:11
   |
32 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `({integer}, {integer})`

error[E0271]: type mismatch resolving `<impl futures_core::stream::Stream as futures_core::stream::Stream>::Item == (i32, i32)`
  --> $DIR/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `(i32, i32)`
   = note: the return type of a function must have a statically known size

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:30:15
   |
30 |         yield 3;
   |               ^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:32:12
   |
32 |     yield (1, 2) //~ ERROR mismatched types
   |            ^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:32:15
   |
32 |     yield (1, 2) //~ ERROR mismatched types
   |               ^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:32:11
   |
32 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:27:1
   |
27 | #[stream(item = (i32, i32))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> $DIR/bad-item-type.rs:24:11
   |
24 |     yield val; //~ ERROR mismatched types
   |           ^^^
   |           |
   |           expected enum `std::option::Option`, found integer
   |           help: try using a variant of the expected enum: `Some(val)`
   |
   = note: expected enum `std::option::Option<_>`
              found type `{integer}`

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:18:9
   |
18 |     let val = Some(42);
   |         ^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:16:1
   |
16 | #[stream(item = Option<i32>, boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:23:9
   |
23 |     let val = val.unwrap();
   |         ^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:16:1
   |
16 | #[stream(item = Option<i32>, boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:24:11
   |
24 |     yield val; //~ ERROR mismatched types
   |           ^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:16:1
   |
16 | #[stream(item = Option<i32>, boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
  --> $DIR/bad-item-type.rs:40:11
   |
40 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ expected integer, found tuple
   |
   = note: expected type `{integer}`
             found tuple `({integer}, {integer})`

error[E0271]: type mismatch resolving `<impl futures_core::stream::Stream as futures_core::stream::Stream>::Item == (i32, i32)`
  --> $DIR/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected tuple, found integer
   |
   = note: expected tuple `(i32, i32)`
               found type `{integer}`
   = note: required for the cast to the object type `dyn futures_core::stream::Stream<Item = (i32, i32)> + std::marker::Send`
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:38:15
   |
38 |         yield 3;
   |               ^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:40:12
   |
40 |     yield (1, 2) //~ ERROR mismatched types
   |            ^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:40:15
   |
40 |     yield (1, 2) //~ ERROR mismatched types
   |               ^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0698]: type inside generator must be known in this context
  --> $DIR/bad-item-type.rs:40:11
   |
40 |     yield (1, 2) //~ ERROR mismatched types
   |           ^^^^^^ cannot infer type for type `{integer}`
   |
note: the type is part of the generator because of this `yield`
  --> $DIR/bad-item-type.rs:35:1
   |
35 | #[stream(item = (i32, i32), boxed)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
