error: unexpected token: `bar`
  --> tests/ui/invalid-argument.rs:12:21
   |
12 |         #[for_await(bar)] //~ ERROR unexpected token
   |                     ^^^

error: unexpected token in attribute
  --> tests/ui/invalid-argument.rs:23:20
   |
23 |         #[for_await(bar)] //~ ERROR unexpected token
   |                    ^

error: unexpected token in attribute
  --> tests/ui/invalid-argument.rs:29:20
   |
29 |         #[for_await()] //~ ERROR unexpected token
   |                    ^

error: unexpected end of input, expected `item`
  --> tests/ui/invalid-argument.rs:37:5
   |
37 |     #[stream] //~ ERROR unexpected end of input, expected `item`
   |     ^^^^^^^^^
   |
   = note: this error originates in the attribute macro `stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `item = <type>`, found `item`
  --> tests/ui/invalid-argument.rs:40:14
   |
40 |     #[stream(item)] //~ ERROR expected `=`
   |              ^^^^

error: expected `item = <type>`, found `item =`
  --> tests/ui/invalid-argument.rs:43:14
   |
43 |     #[stream(item = )] //~ ERROR unexpected end of input, expected one of
   |              ^^^^^^

error: expected `item`
  --> tests/ui/invalid-argument.rs:46:14
   |
46 |     #[stream(baz, item = i32)] //~ ERROR expected `item`
   |              ^^^

error: unexpected argument: baz
  --> tests/ui/invalid-argument.rs:49:26
   |
49 |     #[stream(item = i32, baz)] //~ ERROR unexpected argument
   |                          ^^^

error: expected `item`
  --> tests/ui/invalid-argument.rs:55:14
   |
55 |     #[stream(,item = i32)] //~ ERROR expected `item`
   |              ^

error: expected `,`
  --> tests/ui/invalid-argument.rs:58:25
   |
58 |     #[stream(item = i32 item = i32)] //~ ERROR expected `,`
   |                         ^^^^

error: duplicate `item` argument
  --> tests/ui/invalid-argument.rs:61:26
   |
61 |     #[stream(item = i32, item = i32)] //~ ERROR duplicate `item` argument
   |                          ^^^^^^^^^^

error: duplicate `boxed` argument
  --> tests/ui/invalid-argument.rs:64:33
   |
64 |     #[stream(item = i32, boxed, boxed)] //~ ERROR duplicate `boxed` argument
   |                                 ^^^^^

error: duplicate `boxed_local` argument
  --> tests/ui/invalid-argument.rs:67:39
   |
67 |     #[stream(item = i32, boxed_local, boxed_local)] //~ ERROR duplicate `boxed_local` argument
   |                                       ^^^^^^^^^^^

error: `boxed` and `boxed_local` may not be used at the same time
  --> tests/ui/invalid-argument.rs:70:39
   |
70 |     #[stream(item = i32, boxed_local, boxed)] //~ ERROR `boxed` and `boxed_local` cannot be used at the same time.
   |                                       ^^^^^

error: unexpected end of input, expected `ok`
  --> tests/ui/invalid-argument.rs:77:5
   |
77 |     #[try_stream] //~ ERROR unexpected end of input, expected `ok`
   |     ^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `try_stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of input, expected `ok`
  --> tests/ui/invalid-argument.rs:80:5
   |
80 |     #[try_stream(error = ())] //~ ERROR unexpected end of input, expected `ok`
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `try_stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `ok = <type>`, found `ok`
  --> tests/ui/invalid-argument.rs:83:18
   |
83 |     #[try_stream(ok)] //~ ERROR expected `=`
   |                  ^^

error: expected `ok = <type>`, found `ok =`
  --> tests/ui/invalid-argument.rs:86:18
   |
86 |     #[try_stream(ok = )] //~ ERROR unexpected end of input, expected one of
   |                  ^^^^

error: unexpected end of input, expected `error`
  --> tests/ui/invalid-argument.rs:89:5
   |
89 |     #[try_stream(ok = ())] //~ ERROR unexpected end of input, expected `error`
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the attribute macro `try_stream` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `error = <type>`, found `error`
  --> tests/ui/invalid-argument.rs:92:18
   |
92 |     #[try_stream(error)] //~ ERROR expected `=`
   |                  ^^^^^

error: expected `error = <type>`, found `error =`
  --> tests/ui/invalid-argument.rs:95:18
   |
95 |     #[try_stream(error = )] //~ ERROR unexpected end of input, expected one of
   |                  ^^^^^^^

error: expected `ok`
  --> tests/ui/invalid-argument.rs:98:18
   |
98 |     #[try_stream(baz, ok = (), error = ())] //~ ERROR expected `ok`
   |                  ^^^

error: expected `error`
   --> tests/ui/invalid-argument.rs:101:27
    |
101 |     #[try_stream(ok = (), baz, error = ())] //~ ERROR expected `error`
    |                           ^^^

error: unexpected argument: baz
   --> tests/ui/invalid-argument.rs:104:39
    |
104 |     #[try_stream(ok = (), error = (), baz)] //~ ERROR unexpected argument
    |                                       ^^^

error: expected `ok`
   --> tests/ui/invalid-argument.rs:110:18
    |
110 |     #[try_stream(,ok = () error = ())] //~ ERROR expected `ok`
    |                  ^

error: expected `,`
   --> tests/ui/invalid-argument.rs:113:26
    |
113 |     #[try_stream(ok = () error = ())] //~ ERROR expected `,`
    |                          ^^^^^

error: expected `,`
   --> tests/ui/invalid-argument.rs:116:38
    |
116 |     #[try_stream(ok = (), error = () error = ())] //~ ERROR expected `,`
    |                                      ^^^^^

error: duplicate `ok` argument
   --> tests/ui/invalid-argument.rs:119:27
    |
119 |     #[try_stream(ok = (), ok = (), error = ())] //~ ERROR duplicate `ok` argument
    |                           ^^^^^^^

error: duplicate `ok` argument
   --> tests/ui/invalid-argument.rs:122:39
    |
122 |     #[try_stream(ok = (), error = (), ok = (), error = ())] //~ ERROR duplicate `ok` argument
    |                                       ^^^^^^^

error: duplicate `error` argument
   --> tests/ui/invalid-argument.rs:125:39
    |
125 |     #[try_stream(ok = (), error = (), error = ())] //~ ERROR duplicate `error` argument
    |                                       ^^^^^^^^^^

error: duplicate `boxed` argument
   --> tests/ui/invalid-argument.rs:128:46
    |
128 |     #[try_stream(ok = (), error = (), boxed, boxed)] //~ ERROR duplicate `boxed` argument
    |                                              ^^^^^

error: duplicate `boxed_local` argument
   --> tests/ui/invalid-argument.rs:131:52
    |
131 |     #[try_stream(ok = (), error = (), boxed_local, boxed_local)] //~ ERROR duplicate `boxed_local` argument
    |                                                    ^^^^^^^^^^^

error: `boxed` and `boxed_local` may not be used at the same time
   --> tests/ui/invalid-argument.rs:134:52
    |
134 |     #[try_stream(ok = (), error = (), boxed_local, boxed)] //~ ERROR `boxed` and `boxed_local` cannot be used at the same time.
    |                                                    ^^^^^
