error[E0727]: `async` generators are not yet supported
  --> $DIR/nested.rs:17:13
   |
17 |             yield i * i; //~ ERROR `async` generators are not yet supported [E0727]
   |             ^^^^^^^^^^^

error[E0308]: mismatched types
  --> $DIR/nested.rs:17:19
   |
17 |             yield i * i; //~ ERROR `async` generators are not yet supported [E0727]
   |                   ^^^^^ expected `()`, found `i32`
