error[E0412]: cannot find type `L` in this scope
 --> $DIR/unresolved-type.rs:5:17
  |
5 | #[stream(item = L)] //~ ERROR cannot find type `Left` in this scope
  |                 ^ not found in this scope
6 | async fn foo() {}
  |             - help: you might be missing a type parameter: `<L>`

error[E0698]: type inside generator must be known in this context
 --> $DIR/unresolved-type.rs:5:1
  |
5 | #[stream(item = L)] //~ ERROR cannot find type `Left` in this scope
  | ^^^^^^^^^^^^^^^^^^^ cannot infer type for type parameter `T` declared on the enum `Poll`
  |
note: the type is part of the generator because of this `yield`
 --> $DIR/unresolved-type.rs:5:1
  |
5 | #[stream(item = L)] //~ ERROR cannot find type `Left` in this scope
  | ^^^^^^^^^^^^^^^^^^^
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
